src/configs/transpose-config.c

```diff
@@ -214,10 +214,41 @@ static void init_transpose_config(void) {
       .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x24_transposec_ukernel__1x2_scalar,
       .tile_size = 32,
     };
-    transpose_config.x32 = (struct xnn_transpose_subconfig) {
-      .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__2x4_scalar_int,
-      .tile_size = 32,
-    };
+    #if XNN_ENABLE_RISCV_VECTOR
+      const struct xnn_hardware_config* hardware_config = xnn_init_hardware_config();
+      assert(hardware_config != NULL);
+      if (hardware_config->vlenb >= 128) {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
```

Comment:
having all these rvv kernels pulls in all of them... is this really best?

Reply:
Thank you for the feedback. It is the best choice for each vlenb.